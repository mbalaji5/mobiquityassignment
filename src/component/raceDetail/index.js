export { default as RaceDetail } from "./raceDetail";
