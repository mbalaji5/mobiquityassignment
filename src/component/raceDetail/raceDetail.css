.raceDetail {
  margin: 16px;
  padding: 8px 0px 24px 14px;
  border-bottom: 1px solid;
}

.highlight {
  color: #00afef;
  animation: blinker 1s linear infinite;
}

@keyframes blinker {
  50% {
    opacity: 0;
  }
}
