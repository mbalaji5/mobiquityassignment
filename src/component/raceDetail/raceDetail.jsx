import React from "react";
import PropTypes from "prop-types";
import "./raceDetail.css";
// Component that renders race winner details
// Used functional component.
// Props: race, race result details and season champion id

const RaceDetail = ({
  raceName,
  Results = [],
  championDriverId,
  date,
  round
}) => {
  const {
    Driver: { driverId, familyName, givenName, nationality },
    points,
    Time: { time }
  } = Results[0];
  return (
    <div className="raceDetail">
      <h4>{raceName} :</h4>
      <h4 className={driverId === championDriverId ? "highlight" : ""}>
        {familyName} {givenName}
      </h4>
      <br />
      <span>
        Date : {date} | Round : {round} | Point : {points} | Duration : {time} |
        Nationality : {nationality}
      </span>
    </div>
  );
};

RaceDetail.propTypes = {
  raceName: PropTypes.string.isRequired,
  Results: PropTypes.array.isRequired,
  championDriverId: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  round: PropTypes.string.isRequired
};

RaceDetail.defaultProps = {
  raceName: "Race 1",
  Results: [],
  championDriverId: "Racer1",
  date: "1/1/2021",
  round: 2
};
export default RaceDetail;
