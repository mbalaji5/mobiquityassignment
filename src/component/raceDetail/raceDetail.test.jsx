import React from "react";
import { render, screen } from "@testing-library/react";
import { RaceDetail } from "../raceDetail";

const props = {
  season: 2021,
  round: "1",
  raceName: "Bahrain Grand Prix",
  date: "2021-03-28",
  Results: [
    {
      number: "44",
      position: "1",
      positionText: "1",
      points: "25",
      Driver: {
        driverId: "hamilton",
        permanentNumber: "44",
        code: "HAM",
        url: "http://en.wikipedia.org/wiki/Lewis_Hamilton",
        givenName: "Lewis",
        familyName: "Hamilton",
        dateOfBirth: "1985-01-07",
        nationality: "British"
      },
      Constructor: {
        constructorId: "mercedes",
        url: "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        name: "Mercedes",
        nationality: "German"
      },
      grid: "2",
      laps: "56",
      status: "Finished",
      Time: { millis: "5523897", time: "1:32:03.897" },
      FastestLap: {
        rank: "4",
        lap: "44",
        Time: { time: "1:34.015" },
        AverageSpeed: { units: "kph", speed: "207.235" }
      }
    }
  ],
  championDriverId: "hamilton"
};

describe("Should render champion detail", () => {
  it("Should validate race name", () => {
    render(<RaceDetail {...props} />);
    const element = screen.getByText(/Bahrain Grand Prix/i);
    expect(element).toBeDefined();
  });
  it("Should validate winner full name", () => {
    render(<RaceDetail {...props} />);
    const element = screen.getByText(/Hamilton Lewis/i);
    expect(element).toBeDefined();
  });
  it("Should validate highlight class name if winner is world champion", () => {
    render(<RaceDetail {...props} />);
    const element = screen.getByText(/Hamilton Lewis/i);
    expect(element.classList.contains("highlight")).toBeTruthy();
  });
  it("Should validate winner other details", () => {
    render(<RaceDetail {...props} />);
    const element = screen.getByText(
      /Date : 2021-03-28 | Round : 1 | Point : 25 | Duration : 1:32:03.897 | Nationality : German/i
    );
    expect(element).toBeDefined();
  });
});
describe("Should validate race detail", () => {
  it("Should validate shopshot", () => {
    const { baseElement } = render(<RaceDetail {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
