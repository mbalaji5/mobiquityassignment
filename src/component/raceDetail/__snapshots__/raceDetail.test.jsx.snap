// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render race detail shopshot Should validate shopshot 1`] = `
<body>
  <div>
    <div
      class="raceDetail"
    >
      <h4>
        Bahrain Grand Prix
         :
      </h4>
      <h4
        class="highlight"
      >
        Hamilton
         
        Lewis
      </h4>
      <br />
      <span>
        Date : 
        2021-03-28
         | Round : 
        1
         | Point : 
        25
         | Duration : 
        1:32:03.897
         | Nationality : 
        British
      </span>
    </div>
  </div>
</body>
`;

exports[`Should validate race detail Should validate shopshot 1`] = `
<body>
  <div>
    <div
      class="raceDetail"
    >
      <h4>
        Bahrain Grand Prix
         :
      </h4>
      <h4
        class="highlight"
      >
        Hamilton
         
        Lewis
      </h4>
      <br />
      <span>
        Date : 
        2021-03-28
         | Round : 
        1
         | Point : 
        25
         | Duration : 
        1:32:03.897
         | Nationality : 
        British
      </span>
    </div>
  </div>
</body>
`;
